File: app/src/main/java/com/zhiyong/tingxie/db/QuizPinyin.java
package com.zhiyong.tingxie.db;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Index;
import androidx.room.PrimaryKey;
import androidx.annotation.NonNull;

@Entity(tableName = "quiz_pinyin", indices = {@Index("quiz_id"), @Index("pinyin_string")})
public class QuizPinyin {
    @PrimaryKey(autoGenerate = true)
    private long id;
    @ColumnInfo(name = "quiz_id")
    @NonNull
    private Long quizId;
    @ColumnInfo(name = "pinyin_string")
    @NonNull
    private String pinyinString;
    @ColumnInfo(name = "word_string")
    @NonNull
    private String wordString;
    @NonNull
    private boolean asked;

    public QuizPinyin(long quizId, @NonNull String pinyinString,
                      @NonNull String wordString, @NonNull boolean asked) {
        this.quizId = quizId;
        this.pinyinString = pinyinString;
        this.wordString = wordString;
        this.asked = asked;
    }

    @NonNull
    public Long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    @NonNull
    public Long getQuizId() {
        return quizId;
    }

    public void setQuizId(long quizId) {
        this.quizId = quizId;
    }

    @NonNull
    public String getPinyinString() {
        return pinyinString;
    }

    public void setPinyinString(String pinyinString) {
        this.pinyinString = pinyinString;
    }

    @NonNull
    public String getWordString() {
        return wordString;
    }

    public void setWordString(String wordString) {
        this.wordString = wordString;
    }

    public boolean isAsked() {
        return asked;
    }

    public void setAsked(boolean asked) {
        this.asked = asked;
    }
}

File: app/src/main/java/com/zhiyong/tingxie/db/QuizPinyin.kt
package com.zhiyong.tingxie.db

import androidx.annotation.NonNull
import androidx.room.*

// todo: 11/13/22 Should be able to stop using QuizPinyin when switch to remote.
@Entity(
    tableName = "quiz_pinyin",
    indices = [Index("quiz_id"), Index("pinyin_string")]
)
data class QuizPinyin(
    @PrimaryKey(autoGenerate = true) var id: Long,
    @ColumnInfo(name = "quiz_id") @NonNull val quizId: Long,
    @ColumnInfo(name = "pinyin_string") @NonNull val pinyinString: String,
    @ColumnInfo(name = "word_string") @NonNull val characters: String,
    @NonNull val asked: Boolean,
    @NonNull val status: String = "NOT_DELETED",
    ) {
    @Ignore
    constructor(quizId: Long, pinyinString: String, wordString: String, asked: Boolean):
        this(0, quizId, pinyinString, wordString, asked)
}
